### PR TITLE
Update API client functions - Closes #6099

### LINKED ISSUE #6099: Update api client functions to accept hex string
### Description

Currently, most of the function that requires binary input in APIClient accepts only Buffer, but it should also accept hex string

```
client.block.get
client.block.decode
client.account.get
client.account.decode
client.transaction.get
client.transaction.decode
```

### Motivation

- There were misuse of function reported frequently
- when `subscribe` to event, binary data is returned as hex string

### Acceptance Criteria

- Unit test added
- If input is neither Buffer nor hex string, it should throw an error

### PR DESCRIPTION
**Pull request by Incede**
### What was the problem?

This PR resolves #6099 

### How was it solved?

* Accept hex string as input for `get` & `decode` functions

### How was it tested?

* Added unit tests

### CODE REVIEW

File: elements/lisk-api-client/test/unit/account.spec.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ describe('account', () => {
 			});
 		});
 
-		describe('get', () => {
+		describe('get from address as buffer', () => {
```
Comment (ManuGowda) [inline elements/lisk-api-client/test/unit/account.spec.ts:68]: probably we can structure it like
```
describe('get', () => {
   describe('account by address as string', () => {
      it('should invoke app:getAccount', async () => {});
   });
   
   describe('account by address as buffer', () => {
      it('should invoke app:getAccount', async () => {});
   });
});
```
?
if yes, then same comment for below tests as well.

**Review by ManuGowda — COMMENTED** [review]
LGTM, some minor suggestion

File: elements/lisk-api-client/test/unit/account.spec.ts [hunk 2]
```diff
@@ -66,21 +66,42 @@ describe('account', () => {
...
-			it('should invoke app:getAccount', async () => {
-				// Arrange
-				when(channel.invoke as any)
-					.calledWith('app:getAccount', { address: '9d0149b0962d44bfc08a9f64d5afceb6281d7fb5' })
-					.mockResolvedValue(encodedAccountHex as never);
+			describe('account by address as buffer', () => {
+				it('should invoke app:getAccount', async () => {
+					// Arrange
+					when(channel.invoke as any)
+						.calledWith('app:getAccount', { address: '9d0149b0962d44bfc08a9f64d5afceb6281d7fb5' })
```
Comment (ishantiw) [inline elements/lisk-api-client/test/unit/account.spec.ts:73]: shouldn't address be passed as a buffer in this test case?

File: elements/lisk-api-client/test/unit/block.spec.ts [hunk 3]
```diff
@@ -130,14 +130,29 @@ describe('block', () => {
...
+					// Act
+					await block.get(blockId);
 
-				// Assert
-				expect(channel.invoke).toHaveBeenCalledTimes(1);
-				expect(channel.invoke).toHaveBeenCalledWith('app:getBlockByID', {
-					id: blockId.toString('hex'),
+					// Assert
+					expect(channel.invoke).toHaveBeenCalledTimes(1);
+					expect(channel.invoke).toHaveBeenCalledWith('app:getBlockByID', {
```
Comment (ishantiw) [inline elements/lisk-api-client/test/unit/block.spec.ts:140]: same here, `id` should be a buffer?

**Review by ishantiw — COMMENTED** [review]
A few comments and Lisk Framework Plugins Functional tests are failing

**Review by ishantiw — COMMENTED** [review]
oh ignore my comments, I see that its always converted to hex before calling the channel, only tests are not passing

File: elements/lisk-api-client/test/unit/account.spec.ts (hunk 2 above)
Comment (ishantiw) [reply elements/lisk-api-client/test/unit/account.spec.ts:73]: channel is always called with address hex string so ignore the comment

**Review by ManuGowda — APPROVED** [review]

**Review by ishantiw — APPROVED** [review]

**Pull request merged**